src/matcha_ml/services/analytics_service.py

```diff
@@ -51,6 +85,63 @@ def execute_analytics_event(
     return result, error_code
 
 
+def _time_event(
+    func: Callable[..., Any], *args: Any, **kwargs: Any
+) -> Tuple[Any, Exception, float, float]:
+    """Times the execution of the function decorated by track.
+
+    Args:
+        func (Callable): The function decorated by track.
+        *args (Any): arguments passed to the function.
+        **kwargs (Any): additional keyword arguments passed to the function.
+
+    Returns:
+        Tuple[Any, Exception, float, float]: The result of the call to func, the error code, the start time of execution, the end time of execution.
+    """
+    ts = perf_counter()
+    result, error_code = _execute_analytics_event(func, *args, **kwargs)
+    te = perf_counter()
+
+    return result, error_code, ts, te
+
+
+def _post_event(
+    event_name: AnalyticsEvent,
+    matcha_state_uuid: Optional[str],
+    global_params: GlobalParameters,
+    error_code: Optional[Exception],
+    time_taken: float,
+) -> Tuple[Annotated[bool, "success"], Annotated[str, "message"]]:
+    """Posts the tracked analytics to Segment.
+
+    Args:
+        event_name (AnalyticsEvent): The enumerated name of the analytics event.
+        matcha_state_uuid (str): the uuid for the Matcha state file.
+        global_params (GlobalParameters): The GlobalParameters object containing the user id.
+        error_code (Optional[Exception]): The error propagated by a failing underlying command.
+        time_taken (float): The time taken for the underlying command to execute.
+
+    Returns:
+        A boolean representing the status of the event posting, the message representing the status of the event posting.
+    """
+    client = analytics.Client(WRITE_KEY, max_retries=1, debug=False)
+
+    try:
```

Comment:
If you take a look at the `track()` source code, you'll notice that this function returns the same tuple that you're returning here. So you don't need the `try: ... except: ...` and you can instead _directly_ return the result of  `track()`

Reply:
I see. Good point. Updated.